packages/core/admin/server/services/api-token.js

```diff
@@ -395,6 +410,35 @@ const update = async (id, attributes) => {
   };
 };
 
+/**
+ * Return a layout of the api actions
+ *
+ * @returns {Promise<ApiTokenPermissionSubject[]>}
+ */
+const getApiTokenLayout = async () => {
+  const actions = await strapi.contentAPI.permissions.getActionsMap();
```

Comment:
getActionsMap was mainly written to help get the data in the format needed for the front-end. Instead of mapping again here, maybe it would make sense to change getActionsMap to return in this format? (except for the filter, so we can still use it more generally if we want)

@Convly what do you think?

Reply:
@innerdvations it depends if we see the format that the frontend needs can be used for other stuff, then yeah why not.
but if it doesn't and that will make getActionsMap specific for ApiTokenLayout then I believe it would be best kept as it is

Reply:
Hey, for me `getActionsMap` should stay as it is: a way to build a tree representation of the content APIs actions, so I wouldn't touch its behavior.

Then, regarding the actual route, I don't know if it makes sense (for now) to create a dedicated route to get api-token's permissions. We could keep a generic endpoint to get the content-api's list of available actions you can do.
It could then be used both by API token & U&P to build the frontend layout based on the same data with the same structure.

However, if we know for sure we're going to make some changes in the future on the map structure (either for U&P or API token), then we could abstract the generic logic on a deeper level (such as in a service <=> getActionsMap) and creates distinct endpoint (one for API token & one for U&P), then call the common service in both controllers.
If we want to proceed with the last point, then it's basically really close to what you've done. My only concern then, is the actual format that is returned. I don't really get the point/use of returning those subjectId, label, etc... Instead of building them directly in the frontend.

Reply:
Also, `subjectId` is a bit confusing since, in the permissions, we're using the whole `source::api.controller.action` string to build what we call the action and don't use a subject for them. 

On a final thought, I don't think we need to overengineer this by creating the notion of layout in the backend & instead just return the tree of actions you can do & then build smth from this tree in the frontend. But that's just my opinion. Feel free to debate it :slightly_smiling_face:

Reply:
Just to add some context, in users & permissions, we currently return only the actions/permissions map & format the names to build the accordion.

Here's how the API names are formatted (api::address => Address)

https://github.com/strapi/strapi/blob/features/api-token-v2/packages/plugins/users-permissions/admin/src/utils/formatPluginName.js#L24

Reply:
Looking at it now I generally agree with what you are saying, we can go the simple route on the backend because I know in the frontend we'll be doing layout changes anyways, so I might as well just move everything there.
An it would actually be a cool idea to basically have a content api permissions route for both users & permissions and Api Token since they are basically the same .

Reply:
on that note, I think the best idea is to create a separate route "content-api-permissions", and work with that to build the Api Token layout in the frontend, and perhaps after the api token is done, I'll be able to think about perhaps refactoring the user-permissions plugin to use the same route as well